File: backend/src/main/java/se/migrationsverket/ihpservice/api/rest/v1/dto/NodeIdDto.java
package se.migrationsverket.ihpservice.api.rest.v1.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Builder;
import lombok.Data;

import jakarta.validation.constraints.NotNull;

@Data
@Builder
public class NodeIdDto {
    @NotNull
    Integer id;
    @NotNull
    @Schema(allowableValues = {"issuenode", "documentnode", "processnode", "pgnode", "oanode", "csnode"})
    String nodeName;
    Integer index;
}

File: frontend/src/models/models/NodeIdDto.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

export type NodeIdDto = {
    id: number;
    nodeName: 'issuenode' | 'documentnode' | 'processnode' | 'pgnode' | 'oanode' | 'csnode';
    index?: number;
};
